File: 2016 czerwiec/c++/4.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

long long to_ten(const string& s)
{
    long long ans = 0;
    long long a = 1;
    for (int i = s.size() - 2; i >= 0; i--)
    {
        ans += (s[i] - '0') * a;
        a *= 8;
    }
    return ans;
}

int main()
{
    ifstream in("liczby.txt");
    ofstream out("wyniki_6_4.txt");

    int ans = 0;
    for (int i = 1; i <= 999; i++)
    {
        string s;
        in >> s;
        if (s[s.size() - 1] == '8')
            ans += to_ten(s);
    }
    out << ans;
}

File: 2016 czerwiec/python/4.py
with open("liczby.txt", "r") as f:
    file = f.read().strip().split('\n')

ans = 0
with open("wyniki_6_4.txt", "w") as f:
    for s in file:
        if s[-1] == '8':
            ans += int(s[:-1], 8)
    f.write(str(ans))